### PR TITLE
[Feature][K8S deploy] add helm chart

### PR DESCRIPTION
**Pull request by FrommyMind**
### Purpose of this pull request
#8283 Add helm chart and publish-helm-chart workflow.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Tested by my account.

the `publish-helm-chart` workflow run result.
image

the helm repository.
image

I successfully installed Seatunnel cluster using this helm with version `2.3.8` on my own K8S cluster.

### Check list

* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [x ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/start-v2/kubernetes/helm.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```
+## change
+seatunnel.default.svc.cluster.local
+## to
+seatunnel.<your namespace >.svc.cluster.local
+
+helm install seatunnel . -n <your namespace>
+```
+
+## What's More
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/helm.md:77]: how to submit job?

Comment (FrommyMind) [reply docs/en/start-v2/kubernetes/helm.md:77]: Update doc with how to submit job.

**Comment by flyrun9527** [comment]
helm安装后 开启ingress true 报错 ingress.yaml中 其中port: name 应该是port: number
应该是
spec:
rules:

host: "{{ .Values.ingress.host }}"
http:
paths:
- path: {{ .Values.ingress.path }}
backend:
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
service:
name: {{ include "seatunnel.fullname" . }}-master
port:
number: 5801

**Comment by FrommyMind** [comment]
> helm安装后 开启ingress true 报错 ingress.yaml中 其中port: name 应该是port: number 应该是 spec: rules:
> 
> host: "{{ .Values.ingress.host }}" http: paths:
> 
> * path: {{ .Values.ingress.path }}
>   backend:
>   {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
>   service:
>   name: {{ include "seatunnel.fullname" . }}-master
>   port:
>   number: 5801


Fixed it.

File: .github/workflows/publish-helm-chart.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      packages: write
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set environment variables
+        run: |
+            echo "HUB=registry-1.docker.io/apache" >> $GITHUB_ENV
+            echo "DOCKER_REGISTRY=docker.io" >> $GITHUB_ENV
+            echo "DOCKER_USERNAME=${{ secrets.DOCKERHUB_USER }}" >> $GITHUB_ENV
+            echo "DOCKER_PASSWORD=${{ secrets.DOCKERHUB_TOKEN }}" >> $GITHUB_ENV
```
Comment (Hisoka-X) [inline .github/workflows/publish-helm-chart.yaml]: we can use this way https://github.com/apache/seatunnel/blob/dev/.github/workflows/publish-docker.yaml#L27-L29 to replace it.

File: docs/en/start-v2/kubernetes/helm.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+## Submit Job
+
+The default config doesn't enable ingress, so you need forward the master restapi.
+```bash
+kubectl port-forward -n default svc/seatunnel-master 5801:5801
+```
+Then you can access restapi with "http://127.0.0.1/5801/"
+
+If you want to use ingress, update `value.yaml`
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/helm.md:52]: We should use new rest api port to submit job. Please refer https://github.com/apache/seatunnel/blob/dev/docs/en/seatunnel-engine/rest-api-v2.md#overview

File: .github/workflows/publish-helm-chart.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      packages: write
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set environment variables
+        run: |
+            echo "HUB=registry-1.docker.io/apache" >> $GITHUB_ENV
+            echo "DOCKER_REGISTRY=docker.io" >> $GITHUB_ENV
+            echo "DOCKER_USERNAME=${{ env.DOCKER_USERNAME }}" >> $GITHUB_ENV
+            echo "DOCKER_PASSWORD=${{ env.DOCKER_PASSWORD }}" >> $GITHUB_ENV
```
Comment (Hisoka-X) [inline .github/workflows/publish-helm-chart.yaml]: we can move it into https://github.com/apache/seatunnel/pull/8291/files#diff-6de13ed8c4508bc7e0e6447848811f795b32d0cfbdd96d0642c2896b76796307R27. And delete this part.

File: docs/zh/start-v2/kubernetes/helm.md [hunk 5]
```diff
@@ -75,13 +75,11 @@ MASTER_POD=$(kubectl get po -l  'app.kubernetes.io/name=seatunnel-master' | sed
 # 进入master pod
 kubectl -n default exec -it $MASTER_POD -- /bin/bash
 # 执行 restapi
-curl http://127.0.0.1:5801/hazelcast/rest/maps/running-jobs
-curl http://127.0.0.1:5801/hazelcast/rest/maps/system-monitoring-information
+curl http://127.0.0.1:5801/running-jobs
```
Comment (Hisoka-X) [inline docs/zh/start-v2/kubernetes/helm.md:72]: In the rest api v2. The default port is 8080

Comment (FrommyMind) [reply docs/zh/start-v2/kubernetes/helm.md:72]: Is there a Docker image available that I can use for testing?

File: .github/workflows/publish-helm-chart.yaml (hunk 2 above)
Comment (FrommyMind) [reply .github/workflows/publish-helm-chart.yaml]: Fix

File: docs/zh/start-v2/kubernetes/helm.md (hunk 5 above)
Comment (Hisoka-X) [reply docs/zh/start-v2/kubernetes/helm.md:72]: The default docker images support this feature. Just add bind 8080 port to localhost is enough.

File: .github/workflows/publish-helm-chart.yaml (hunk 4 above)
Comment (FrommyMind) [reply .github/workflows/publish-helm-chart.yaml]: OK

File: docs/zh/start-v2/kubernetes/helm.md (hunk 5 above)
Comment (FrommyMind) [reply docs/zh/start-v2/kubernetes/helm.md:72]: Do I need to replace all port "5801" with "8080" in all template files?

Comment (Hisoka-X) [reply docs/zh/start-v2/kubernetes/helm.md:72]: Yes, please. Then please verify it in your enviorment. Thanks.

Comment (FrommyMind) [reply docs/zh/start-v2/kubernetes/helm.md:72]: Ok

Comment (FrommyMind) [reply docs/zh/start-v2/kubernetes/helm.md:72]: Updated.

**Comment by FrommyMind** [comment]
Update with two port, 5801 for cluster join, and 8080 for restapi。
After install， svc as bellow
image

And endpoints as bellow.
image

**Comment by FrommyMind** [comment]
After #8312 merge， need update metric api version

**Comment by liunaijie** [comment]
LGTM, wait for CI pass

**Comment by FrommyMind** [comment]
CI https://github.com/FrommyMind/seatunnel/actions/runs/12476135067 rerun succeed。

**Review by liunaijie — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**